_posts/news/2020-02-19-announcing-scalajs-1.0.0.md

```diff
@@ -0,0 +1,588 @@
+
+## Upgrade to 1.0.0 from 0.6.32 or later
+
+These instructions apply to sbt users.
+If you are using a different build tool, refer to that build tool's documentation for Scala.js support.
+
+As a first approximation, all you need to do is to update the version number in `project/plugins.sbt`:
+
+{% highlight scala %}
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.0.0")
+{% endhighlight %}
+
+In addition, if you use some of the components that have been moved to separate repositories, you will need to add some more dependencies in `project/plugins.sbt`:
+
+If you use `scalajs-stubs`:
+
+* Change its version number to `"1.0.0"`
+
+If you use `jsDependencies` (or rely on the `jsDependencies` of your transitive dependencies):
+
+* Add `addSbtPlugin("org.scala-js" % "sbt-jsdependencies" % "1.0.0")` in `project/plugins.sbt`
+* Add `.enablePlugins(JSDependenciesPlugin)` to Scala.js `project`s
+* Add `.jsConfigure(_.enablePlugins(JSDependenciesPlugin))` to `crossProject`s
+
+If you use the Node.js with jsdom environment (`JSDOMNodeJSEnv`):
+
+* Add `libraryDependencies += "org.scala-js" %% "scalajs-env-jsdom-nodejs" % "1.0.0"` in `project/plugins.sbt`
+
+If you use the PhantomJS environment (`PhantomJSEnv`):
+
+* Add `addSbtPlugin("org.scala-js" % "sbt-scalajs-env-phantomjs" % "1.0.0")` in `project/plugins.sbt`
+
+If you use the Selenium environment (`SeleniumJSEnv`):
+
+* Change its version number to `"1.0.0"`
+
+**Note:** All the above version numbers coincide at `"1.0.0"` for this release, but they will independently evolve in the future.
+Therefore, do not "DRY up" your build by using a common constant (such as `scalaJSVersion`) for those dependencies, or your build will break when updating one of them in the future.
+
+Finally, if your build has
+
+{% highlight scala %}
+scalacOptions += "-P:scalajs:sjsDefinedByDefault"
+{% endhighlight %}
+
+you will need to remove it (Scala.js 1.x always behaves as if `sjsDefinedByDefault` were present).
+
+This should get your build up to speed to Scala.js 1.0.0.
+From there, you should be able to test whether things go smoothly, or whether you are affected by the breaking changes detailed below.
+
+## Breaking changes
+
+This section discusses the backward incompatible changes, which might affect your project.
+
+### sbt 0.13.x is not supported anymore
+
+You will not be able to use Scala.js 1.0.0 with sbt 0.13.x.
+You will have to upgrade to sbt 1.2.1 or later.
+As of this writing, the latest version is 1.3.8.
+
+### Scala 2.10.x, 2.11.{0-11} and 2.12.0 are not supported anymore
+
+The title says it all: you cannot use Scala.js anymore with any of:
+
+{% highlight scala %}
+scalaVersion := "2.10.x" // for any x
+scalaVersion := "2.11.x" // for 0 <= x <= 11
+scalaVersion := "2.12.0"
+{% endhighlight %}
+
+* Scala 2.10.x is not supported at all,
+* Only 2.11.12 is still supported in the 2.11.x series, and
+* 2.12.1 and following are supported in the 2.12.x series, but not 2.12.0.
```

Comment:
This might be a useful thing to mention in "before you upgrade": Upgrading the Scala version can (and should) be done before upgrading Scala.js to minimize simultaneous changes.